### PR TITLE
chore: remove shadow write of new sip 68 dataset models

### PR DESCRIPTION
**Pull request by eschutho**
### SUMMARY
We're seeing some sqlalchemy errors related to the dual writes of the sip68 models, so I'm removing the dual writing so that we can decide whether to move forward again and do another backfill migration or start over smaller with one model at a time. We also haven't been writing permissions to the models, so we would need to run a migration for that in the future regardless of whether we stopped dual writing or not, so it seemed to make sense to take a step back and stop the errors for now. 

I also fixed a test on dataset deletion due to sqlalchemy not having access to the database relationship in the session. The fix was to eagerly load the database with the dataset on lookup. I found other logs of this happening for this command, and fixed it here for the tests to pass. There was a query in the after update hook that I removed that was re-loading the database into the session as an unexpected side-effect.

### TESTING INSTRUCTIONS
Normal dataset crud operations should work as normal.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/datasets/dao.py [hunk 1]
```diff
@@ -35,6 +37,26 @@ class DatasetDAO(BaseDAO):  # pylint: disable=too-many-public-methods
     model_cls = SqlaTable
     base_filter = DatasourceFilter
 
+    @classmethod
+    def find_by_ids(cls, model_ids: Union[List[str], List[int]]) -> List[SqlaTable]:
```
Comment (ktmud) [inline superset/datasets/dao.py:41]: Is this related?

**Review by ktmud — APPROVED** [review]
@betodealmeida definitely is more qualified of reviewing this than me, but I agree the dual-write is probably not ideal if we aren't going to resume working on the new models any time soon.

If it's something you are allow to share, can you elaborate what were the errors you saw?

File: superset/datasets/dao.py (hunk 1 above)
Comment (betodealmeida) [reply superset/datasets/dao.py:41]: Looks unrelated... maybe this was supposed to be in another branch?

**Review by betodealmeida — APPROVED** [review]

File: superset/datasets/dao.py (hunk 1 above)
Comment (eschutho) [reply superset/datasets/dao.py:41]: I also mentioned it in the PR description, but yes tests were failing on dataset deletion due to sqlalchemy not having access to the database relationship in the session. I found other logs of this happening for this command, and fixed it here for the tests to pass. There was a query in the after update hook that I removed that was re-loading the database into the session as an unexpected side-effect.

**Comment by john-bodley** [comment]
@betodealmeida and @eschutho what’s the status of SIP-68? Are there blockers for preventing us from reading from the new datamodel tables?

**Comment by eschutho** [comment]
@john-bodley after this change, the new datasets/models and the sqlaTable/columns, etc will be out of sync. Prior to this change, some of the permissions may have been out of sync as well. We're going to try this again, but maybe with a model by model approach so that we can maybe knock out some of the new models quicker and then do the dataset change last, which will be the most complex. We just haven't had much time to contribute to wrapping this up unfortunately.
